packages/daemon/src/daemon.js

```diff
@@ -972,6 +973,14 @@ const makeDaemonCore = async (
     );
   };
 
+  /**
+   * @type {import('./types.js').DaemonCore['incarnateWorker']}
+   */
+  const incarnateWorker = async () => {
+    const formulaNumber = await formulaGraphMutex.enqueue(randomHex512);
+    return incarnateNumberedWorker(formulaNumber);
```

Comment:
I’m unsure whether this is sufficiently synchronized since `randomHex512` will complete and release the lock before `incarnateNumberedWorker` has an opportunity to add the number to the table.

Reply:
Hm, I think that concern applies to all extant uses of `formulaGraphMutex`. Looking at the implementation of the mutex, we synchronously call `unlock()` in a `finally` block after a `return await asyncFn()` (in this case it would be `randomHex512`). Shouldn't control be relinquished to L980 here before anything else that was waiting on the lock?

I could try to write a test for this.

Reply:
#2122